integration/stability/agent_stability_test.sh

```diff
@@ -30,24 +28,26 @@ start_time=$(date +%s)
 end_time=$((start_time+timeout))
 
 function setup {
-	clean_env
-	docker run --runtime=$RUNTIME -d --name $CONTAINER_NAME $IMAGE $PAYLOAD_ARGS
+	sudo systemctl restart containerd
+	clean_env_ctr
+	CONTAINERD_RUNTIME="io.containerd.kata.v2"
+	sudo ctr image pull $IMAGE
+	sudo ctr run --runtime=$CONTAINERD_RUNTIME -d $IMAGE $CONTAINER_NAME sh -c $PAYLOAD_ARGS
 }
 
 function exec_loop {
-	docker exec $CONTAINER_NAME sh -c "echo 'hello world' > file"
-	docker exec $CONTAINER_NAME sh -c "rm -rf /file"
-	docker exec $CONTAINER_NAME sh -c "ls /etc/resolv.conf 2>/dev/null " | grep "/etc/resolv.conf"
-	docker exec $CONTAINER_NAME sh -c "touch /tmp/execWorks"
-	docker exec $CONTAINER_NAME sh -c "ls /tmp | grep execWorks"
-	docker exec $CONTAINER_NAME sh -c "rm -rf /tmp/execWorks"
-	docker exec $CONTAINER_NAME sh -c "ls /etc/foo" || echo "Fail expected"
-	docker exec $CONTAINER_NAME sh -c "cat /tmp/one" || echo "Fail expected"
-	docker exec $CONTAINER_NAME sh -c "exit 42" || echo "Fail expected"
+	sudo ctr t exec --exec-id 1 $CONTAINER_NAME sh -c "echo 'hello world' > file"
+	sudo ctr t exec --exec-id 1 $CONTAINER_NAME sh -c "rm -rf /file"
+	sudo ctr t exec --exec-id 1 $CONTAINER_NAME sh -c "touch /tmp/execWorks"
+	sudo ctr t exec --exec-id 1 $CONTAINER_NAME sh -c "ls /tmp | grep execWorks"
+	sudo ctr t exec --exec-id 1 $CONTAINER_NAME sh -c "rm -rf /tmp/execWorks"
+	sudo ctr t exec --exec-id 1 $CONTAINER_NAME sh -c "ls /etc/foo" || echo "Fail expected"
+	sudo ctr t exec --exec-id 1 $CONTAINER_NAME sh -c "cat /tmp/one" || echo "Fail expected"
+	sudo ctr t exec --exec-id 1 $CONTAINER_NAME sh -c "exit 42" || echo "Fail expected"
 }
```

Comment:
There is a lot of repeated code here. Why not use a variable for the part of the command that is identical every line?

(Bonus: You could leave the variant of the command that uses docker, making it easier to switch).